openml/datasets/functions.py

```diff
@@ -359,6 +361,57 @@ def get_dataset(dataset_id):
     return dataset
 
 
+def attributes_arff_from_df(df):
+    """Create the attributes as specified by the ARFF format using a dataframe.
+
+    Arguments:
+        df : DataFrame, shape (n_samples, n_features)
+            The dataframe containing the data set.
+
+    Returns:
+        attributes_arff : str
+            The data set attributes as required by the ARFF format.
+    """
+    PD_DTYPES_TO_ARFF_DTYPE = {
+        'integer': 'INTEGER',
+        'floating': 'REAL',
+        'string': 'STRING'
+    }
+    attributes_arff = []
+    for column_name in df:
+        # skipna=True does not infer properly the dtype. The NA values are
+        # dropped before the inference instead.
+        column_dtype = pd.api.types.infer_dtype(df[column_name].dropna())
+
+        if column_dtype == 'categorical':
```

Comment:
I missed the discussion in the thread. is this a regular pandas data type? Is there any reason to not use the dtype str? 

(the comments below do not seem to help me too much, sorry)

Reply:
This is an available dtype in pandas to mention that a column is a categorical column.

> Is there any reason to not use the dtype str?

Actually I don't see why categories should be string :). Right now, the liac-arff does not allow otherwise, we could actually think about converting silently the categories to string if a column is of category dtype.

Reply:
Cool, thanks for the clarification :)